src/dbus_mpris.rs

```diff
@@ -161,15 +161,7 @@ async fn create_dbus_server(
     });
 
     let path: Cow<str> = if unique_mpris {
-        format!(
-            "{a}.{b}",
-            a = DBUS_PATH,
-            b = device_name
-                .chars()
-                .filter(|c| c.is_ascii())
-                .collect::<String>()
-        )
-        .into()
+        format!("{a}.instance{b}", a = DBUS_PATH, b = std::process::id()).into()
```

Comment:
This could totally just be the default behavior. There's no need to have config option. Single instance users lose nothing and you gain multi-instance support out of the box. It's a -0 +1 situation.

Reply:
How?

Reply:
Any software that interfaces via dbus would no longer work until they make that change.

Reply:
`dbus-send --print-reply --dest=org.freedesktop.DBus  /org/freedesktop/DBus org.freedesktop.DBus.ListNames | grep org.mpris.MediaPlayer2.spotifyd.instance`

Reply:
> Any software that interfaces via dbus would no longer work until they make that change.

No one should be writing software designed to work with just one specific MPRIS interface. That's broken by design in the 1st place.

Reply:
By "software" ofc you mean hacky bash scripts,lol!!!

Reply:
I plan on adding a command API to librespot similar to [librespot-java's](https://github.com/librespot-org/librespot-java/tree/dev/api) and I've got a pending PR that greatly expands the usefulness of the events (which are trivial to UDP multicast with a few lines of python) so abusing MPRIS in that way will not be necessary in the future.

Reply:
Another option is to check to see if `org.mpris.MediaPlayer2.spotifyd` already exists, if it doesn't just use that name if not use `org.mpris.MediaPlayer2.spotifyd.instance{pid}` That would satisfy both backwards compatibility for existing scripts and the unique name requirement. You also have to un-own the name on drop though to make sure not to orphan it.

Reply:
Thanks to both of you for the various points. I'm personally the author of such a “software” that currently relies on the MPRIS name being exactly the one that it is now. In my case, though, it should be quite trivial to adapt that to the PID behaviour, since I'm listening for new bus names anyway.

From what I read in the various issues in this repo, there are a bunch of people who are using such scripts to control their `spotifyd` instances and as such rely on the name being always the same. If you consider, however, how easy it should be to add support for the new bus names to their scripts, this breakage might be acceptable?

To do this, you could:

- use `playerctl`, that Just Works™ with the new behaviour, even if you specify `playerctl -p spotifyd ...`
- use one of the methods outlined [here](https://unix.stackexchange.com/a/46309) to get a list of all bus names and filter that with `grep "org.mpris.MediaPlayer2.spotifyd"` to get the bus name

If we decide to unconditionally change that, it might still be good to note that and possible solutions somewhere.

Reply:
@eladyn you could easily just use the grep method or similar in playerctl so a user would not need to know the full busname and if it matches multiple instances the throw an an error that the user needs to be more specific.

Reply:
Yeah, what I meant to say with `playerctl` is that when I do:

```
playerctl -p spotifyd status
```
it works out of the box, even if the bus name is e.g. `org.mpris.MediaPlayer2.spotifyd.instance9928`. So there would be no change needed.